File: app-presentation/src/main/java/com/zxjdev/smile/presentation/authorization/register/RegisterContract.java
package com.zxjdev.smile.presentation.authorization.register;

import com.zxjdev.smile.presentation.common.base.BasePresenter;
import com.zxjdev.smile.presentation.common.base.BaseView;

public interface RegisterContract {

  interface View extends BaseView {

    void navigateToMain();

    void initUserComponent();
  }

  interface Presenter extends BasePresenter<View> {

    void handleRegister(String username, String password);
  }
}

File: app-presentation/src/main/java/com/zxjdev/smile/presentation/authorization/register/RegisterContract.kt
package com.zxjdev.smile.presentation.authorization.register

import com.zxjdev.smile.presentation.common.base.BasePresenter
import com.zxjdev.smile.presentation.common.base.BaseView

interface RegisterContract {

    interface View : BaseView {

        fun navigateToMain()

        fun initUserComponent()
    }

    interface Presenter : BasePresenter<View> {

        fun handleRegister(username: String, password: String)
    }
}